hrms/payroll/doctype/bulk_salary_structure_assignment/bulk_salary_structure_assignment.js

```diff
@@ -0,0 +1,325 @@
+// Copyright (c) 2024, Frappe Technologies Pvt. Ltd. and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on("Bulk Salary Structure Assignment", {
+	setup(frm) {
+		frm.trigger("set_queries");
+		frm.trigger("setup_filter_group");
+	},
+
+	async refresh(frm) {
+		frm.trigger("set_primary_action");
+		await frm.trigger("set_payroll_payable_account");
+		frm.trigger("get_employees");
+	},
+
+	from_date(frm) {
+		frm.trigger("get_employees");
+	},
+
+	async company(frm) {
+		await frm.trigger("set_payroll_payable_account");
+		frm.trigger("get_employees");
+	},
+
+	branch(frm) {
+		frm.trigger("get_employees");
+	},
+
+	department(frm) {
+		frm.trigger("get_employees");
+	},
+
+	employment_type(frm) {
+		frm.trigger("get_employees");
+	},
+
+	designation(frm) {
+		frm.trigger("get_employees");
+	},
+
+	grade(frm) {
+		frm.trigger("get_employees");
+	},
+
+	set_primary_action(frm) {
+		frm.disable_save();
+		frm.page.set_primary_action(__("Assign Structure"), () => {
```

Comment:
To make the UX even better, we can also warn about validations beforehand. Eg: 
- Base is required almost every time. Just warn the user that you have not set the base for these rows. Want to proceed?
- Also `validate_dates` function in salary structure assignment checks if we are trying to assign a structure before DOJ or after relieving. Maybe flag such rows - color-code/gray out with a hover message on why structure cannot be assigned to them. Or other option is don't fetch such employees at all